src/core/ext/transport/chttp2/transport/hpack_encoder.h

```diff
@@ -41,17 +42,10 @@
 extern grpc_core::TraceFlag grpc_http_trace;
 
 struct grpc_chttp2_hpack_compressor {
-  uint32_t max_table_size;
-  uint32_t max_table_elems;
-  uint32_t cap_table_elems;
-  /** maximum number of bytes we'll use for the decode table (to guard against
-      peers ooming us by setting decode table size high) */
+  // maximum number of bytes we'll use for the decode table (to guard against
+  // peers ooming us by setting decode table size high)
   uint32_t max_usable_size;
-  /* one before the lowest usable table index */
-  uint32_t tail_remote_index;
-  uint32_t table_size;
-  uint32_t table_elems;
-  uint16_t* table_elem_size;
+  grpc_core::ManualConstructor<grpc_core::HPackEncoderTable> table;
```

Comment:
non-blocking comment - is there a clean way to avoid using `ManualConstructor`?

Reply:
see next commit